webui/react/src/components/kit/InlineForm.tsx

```diff
@@ -0,0 +1,120 @@
+import { Form, FormProps } from 'antd';
+import { Rule } from 'antd/es/form';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import Button from 'components/kit/Button';
+import Icon from 'components/kit/Icon';
+import { Primitive } from 'components/kit/internal/types';
+
+import css from './InlineForm.module.scss';
+
+interface Props extends React.PropsWithChildren, Omit<FormProps, 'children'> {
+  label?: string;
+  displayValue?: Primitive;
+  initialValue?: Primitive;
+  onSubmit: (inputValue: string | number) => Promise<void | Error> | void;
+  required?: boolean;
+  isPassword?: boolean;
+  rules?: Rule[];
+  testId?: string;
+}
+
+const InlineForm: React.FC<Props> = ({
+  label,
+  children,
+  displayValue = '',
```

Comment:
Reiterating here for posterity that this needs to support being a controlled component or an uncontrolled component just like any other input. Right now it's a weird mix of both.